### PR TITLE
Use clusterIP for local services

### LINKED ISSUE #355: Use ClusterIP for local services
When exported service is present in local cluster and globalnet is enabled, return the clusterIP of service instead of GlobalIP.

### LINKED ISSUE #365: [E2E] Test service-discovery prefers local cluster before remote cluster
Please add E2E test for feature https://github.com/submariner-io/lighthouse/issues/331

### PR DESCRIPTION
**Pull request by vthapar**
Fixes #355
Fixes #365 

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

**Comment by vthapar** [comment]
WiP

File: pkg/service/controller.go
Comment (tpantelis) [inline pkg/service/controller.go:70]:
```suggestion
func (c *Controller) GetIP(name, namespace string) (string, bool) {
```

File: pkg/service/controller.go
Comment (tpantelis) [inline pkg/service/controller.go:75]: We should log the error.

File: pkg/serviceimport/map.go
Comment (tpantelis) [inline pkg/serviceimport/map.go:57]: The named return vars are not used. Are you naming them just for documentation?

File: pkg/serviceimport/map_test.go
Comment (tpantelis) [inline pkg/serviceimport/map_test.go:43]: I think we should verify `isLocal` and add a test for it.

File: pkg/serviceimport/map.go
Comment (tpantelis) [inline pkg/serviceimport/map.go:77]:
```suggestion
		return ip, found, cluster == localCluster
```

File: plugin/lighthouse/handler.go
Comment (tpantelis) [inline plugin/lighthouse/handler.go:120]: The OR clause isn't needed here with the suggested change  `serviceImports.GetIP`.

File: plugin/lighthouse/handler.go
Comment (tpantelis) [inline plugin/lighthouse/handler.go:123]: Perhaps it would cleaner to incorporate `localServices` into `serviceImports.GetIP`. But we can consider that for a separate PR.

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/serviceimport/map.go
Comment (vthapar) [reply pkg/serviceimport/map.go:57]: Yes, for documentation.

File: plugin/lighthouse/handler.go
Comment (vthapar) [reply plugin/lighthouse/handler.go:123]: ServiceImport and Service are using two different informers. I have an idea on how to make it work, but will revisit it later after 0.8.0 With the plan to use EndpointSlice instead of ServiceImport for clusterIP, it will become easier.

Comment (tpantelis) [reply plugin/lighthouse/handler.go:123]: `LocalServices` is an interface so we could pass it into `serviceimport.Map`.

Comment (vthapar) [reply plugin/lighthouse/handler.go:123]: Aha. Can do it in a separate PR later.

File: pkg/serviceimport/map_test.go
Comment (vthapar) [reply pkg/serviceimport/map_test.go:43]: Will have to add a new test for it. Just adding an Expect for it here breaks the tests. Can add it later along with GetIP rework.

File: Dockerfile.dapper718299099
Comment (aswinsuryan) [inline Dockerfile.dapper718299099:1]: This file needs to be removed.

**Review by aswinsuryan — COMMENTED** [review]
LGTM

File: Dockerfile.dapper718299099
Comment (vthapar) [reply Dockerfile.dapper718299099:1]: Argh, forgot to cleanup before updating. Will do so when I rebase.

**Review by aswinsuryan — APPROVED** [review]

**Comment by nyechiel** [comment]
@vthapar are you still working on this for 0.8?

**Comment by vthapar** [comment]
> @vthapar are you still working on this for 0.8?

Review comments are already addressed, waiting for next batch of review/approvals.

**Review by tpantelis — APPROVED** [review]

**Pull request merged**
